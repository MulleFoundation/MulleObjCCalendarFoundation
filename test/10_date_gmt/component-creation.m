#ifdef __MULLE_OBJC__
# import <MulleObjCCalendarFoundation/MulleObjCCalendarFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif


//
// the values in stdout were produced by macOS
//
NSTimeInterval   intervals[] =
{
   -32535172800.0,
   -13197686400.0,
   -13197600000.0,
   -13197000000.0,
   -10000000000.0,
   -7299138600.0,
   -4818892320.0,
   -978264000.0,
   -31579200.0,
   -1.0,
   0.0,
   604713600.0,
   604713599.0,
   604627200.0, // = 29.2.2020 0:00:00 { e:1 q:1 wd:7 wO:5 wY:9 wM:5 ns:0 }
   604627199.0,
   575694000.0,
   575693999.0,
   575690400.0,
   575690399.0
};


int   main( void)
{
   NSCalendar        *calendar;
   NSDateComponents  *components;
   NSDate            *date;
   NSTimeInterval    *p;
   NSTimeInterval    *sentinel;

   calendar   = [NSCalendar calendarWithIdentifier:NSGregorianCalendar];
   [calendar setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"GMT"]];
   [calendar setFirstWeekday:2];

   p        = &intervals[ 0];
   sentinel = &p[ sizeof( intervals) / sizeof( intervals[ 0])];
   for( ;p < sentinel; ++p)
   {
      date       = [NSDate dateWithTimeIntervalSinceReferenceDate:*p];
      components = [calendar components:~0
                               fromDate:date];

      printf( "%.1f = %ld.%ld.%ld %ld:%02ld:%02ld.%09ld\n",
            [date timeIntervalSinceReferenceDate],
            (long) [components day], (long) [components month], (long) [components year],
            (long) [components hour], (long) [components minute], (long) [components second],
            (long) [components nanosecond]);
   }
   return( 0);
}