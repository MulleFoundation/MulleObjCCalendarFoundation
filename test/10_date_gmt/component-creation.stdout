-32535172800.0 = 1.1.970 12:00:00.000000000
-13197686400.0 = 14.10.1582 0:00:00.000000000
-13197600000.0 = 15.10.1582 0:00:00.000000000
-13197000000.0 = 21.10.1582 22:40:00.000000000
-10000000000.0 = 11.2.1684 6:13:20.000000000
-7299138600.0 = 13.9.1769 5:30:00.000000000
-4818892320.0 = 18.4.1848 18:48:00.000000000
-978264000.0 = 1.1.1970 12:00:00.000000000
-31579200.0 = 1.1.2000 12:00:00.000000000
-1.0 = 31.12.2000 23:59:59.000000000
0.0 = 1.1.2001 0:00:00.000000000
604713600.0 = 1.3.2020 0:00:00.000000000
604713599.0 = 29.2.2020 23:59:59.000000000
604627200.0 = 29.2.2020 0:00:00.000000000
604627199.0 = 28.2.2020 23:59:59.000000000
575694000.0 = 31.3.2019 3:00:00.000000000
575693999.0 = 31.3.2019 2:59:59.000000000
575690400.0 = 31.3.2019 2:00:00.000000000
575690399.0 = 31.3.2019 1:59:59.000000000
